From b28c566d38c13352c4d97b4f804b23d491df038a Mon Sep 17 00:00:00 2001
From: Nils Kneuper <crazy-ivanovic@gmx.net>
Date: Sat, 25 Oct 2025 01:12:27 +0200
Subject: [PATCH] Changed CMake to not directly pull in boost-system

fixing build issues on archlinux using boost-libs >1.89.0-2

* asturm 2026-01-19: Backported to 1.18 branch

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index afca54a1a78c7..4ea33aa37332b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,7 +52,7 @@
 option(ENABLE_TESTS "Build unit tests")
 option(ENABLE_NLS "Enable building of translations" ${ENABLE_GAME})
 
-set(BOOST_VERSION "1.67")
+set(BOOST_VERSION "1.70")
 
 if(NOT WIN32)
 	set(Lua_FIND_VERSION_MAJOR 5)
@@ -93,7 +93,7 @@
 	find_library(SECURITY_LIBRARY Security REQUIRED)
 endif()
 
-find_package(Boost ${BOOST_VERSION} REQUIRED COMPONENTS iostreams program_options regex system thread random coroutine locale filesystem graph)
+find_package(Boost ${BOOST_VERSION} REQUIRED COMPONENTS iostreams program_options regex thread random coroutine locale filesystem graph)
 find_package(ICU REQUIRED COMPONENTS data i18n uc)
 
 # no, gettext executables are not required when NLS is deactivated
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -248,7 +248,6 @@
 		Boost::iostreams
 		Boost::program_options
 		Boost::regex
-		Boost::system
 		Boost::random
 		Boost::coroutine
 		Boost::locale
@@ -295,7 +294,6 @@
 		Boost::iostreams
 		Boost::program_options
 		Boost::regex
-		Boost::system
 		Boost::random
 		Boost::coroutine
 		Boost::locale
@@ -344,7 +342,6 @@
 		Boost::iostreams
 		Boost::program_options
 		Boost::regex
-		Boost::system
 		Boost::random
 		Boost::coroutine
 		Boost::locale
@@ -385,7 +382,6 @@
 		Boost::iostreams
 		Boost::program_options
 		Boost::regex
-		Boost::system
 		Boost::random
 		Boost::coroutine
 		Boost::locale
